odilia/src/tower.rs

```diff
@@ -405,15 +423,16 @@ where
 // S: is some state type that implements Clone
 // Fut: is a future whoes output is Result<Response, Error>, and which is sendable across threads
 // statically
-impl<F, Fut, S, T1, T2, E> Handler<(Request, T1, T2), S, E> for F
+impl<F, Fut, S, T1, T2, E, R> Handler<(Request, T1, T2), S, E> for F
 where
 	F: FnOnce(E, T1, T2) -> Fut + Clone + Send,
-	Fut: Future<Output = Result<Response, Error>> + Send + 'static,
+	Fut: Future<Output = Result<R, Error>> + Send + 'static,
 	S: Clone,
 	T1: From<S>,
 	T2: From<S>,
+	R: IntoCommands,
 {
-	type Response = Response;
+	type Response = R;
 	type Future = Fut;
 	fn call(self, req: E, state: S) -> Self::Future {
 		self(req, state.clone().into(), state.clone().into())
```

Comment:
why is the cloning of state twice required? maybe wrap state in arc at all times, then use arc::clone here?

Reply:
The `Clone` is expected to be an `Arc<S>` where `S is ScreenReaderState`. The clone should be "cheap". This is how Axum works.